aggregator.py

```
@@ -0,0 +1,98 @@
+  best to send along to the BeliefManager. """
+
+  def __init__(self, mav_queue, radio_queue, belief_box):
+    """ Args:
+      mav_queue: The queue for receiving autopilot data.
+      radio_queue: The queue for receiving radio data.
+      belief_box: The mailbox for sending data to the BeliefManager process. """
+    self.__limitter = PhasedLoopLimitter(BeliefManager.CYCLE_LENGTH)
+
+    self.__mav_queue = mav_queue
+    self.__radio_queue = radio_queue
+    self.__belief_box = belief_box
+
+    self.__clear()
+
+  def __clear(self):
+    """ Clears all the accumulators and buffers. """
+    # Buffer for storing radio data.
+    self.__radio_buffer = []
+
+    # Accumulators for various positional measurements.
+    self.__latitude = 0.0
+    self.__longitude = 0.0
+    self.__x_vel = 0.0
+    self.__y_vel = 0.0
+    self.__message_count = 0
+
+  def iterate(self):
+    """ Runs a single iteration of the process. This should be synched with the
+    frequency at which BeliefManager is run. """
+    logger.debug("Running iteration.")
+    # Clear everything.
+    self.__clear()
+
+    # Get and buffer radio data.
+    while True:
+      try:
+        radio_message = self.__radio_queue.get_nowait()
+      except Queue.Empty:
+        # Nothing more to read.
+        break
+
+      radio_tuple = (radio_message.lob, radio_message.strength)
+      self.__radio_buffer.append(radio_tuple)
+
+    if not self.__radio_buffer:
+      logger.warning("Got no new radio data.")
+
+    # Get and process drone position data.
+    while True:
+      try:
+        mav_message = self.__mav_queue.get_nowait()
+      except Queue.Empty:
+        # Nothing more to read.
+        break
+
+      self.__latitude += mav_message.latitude
+      self.__longitude += mav_message.longitude
+      self.__x_vel += mav_message.velocity_array[0]
+      self.__y_vel += mav_message.velocity_array[1]
+      self.__message_count += 1
```

The maximum float in Python is 1.7976931348623157e+308, so I sort of doubt that that will be a problem. (Obviously, if you want reasonable precision, it's less, but still.) Either way, I don't think the numbers that we're working with here are *that* huge. The only way I could see this being a problem is if for some reason the autopilot spits out an obscene amount of data, in which case, we probably have a larger issue, so I think that we want it to fail loudly.